lib/bolt/inventory/inventory2.rb

```diff
@@ -157,23 +130,26 @@ def update_target(target)
           data['config'] = {}
         end
 
-        data = Bolt::Inventory.localhost_defaults(data) if target.name == 'localhost'
-        # These should only get set from the inventory if they have not yet
-        # been instantiated
-        set_vars_from_hash(target.name, data['vars']) unless @target_vars[target.name]
-        set_facts(target.name, data['facts']) unless @target_facts[target.name]
-        data['features']&.each { |feature| set_feature(target, feature) } unless @target_features[target.name]
-        unless @target_plugin_hooks[target.name]
-          set_plugin_hooks(target.name, @config.plugin_hooks.merge(data['plugin_hooks'] || {}))
+        # Add defaults for special 'localhost' target (currently just config and features)
+        if target.name == 'localhost'
+          data = Bolt::Inventory.localhost_defaults(data)
+          data['features'].each do |feature|
```

Comment:
```suggestion
          data['features']&.each do |feature|
```

Reply:
Actually I think the only time this is nil is if it's *explicitly* set to nil. In that case I think we can remove line 126 here, since it doesn't seem like the group data will ever be nil, since all nodes belong to at least the 'all' group.